src/manipulation.js

```diff
@@ -17,7 +17,7 @@ define([
 
 var
 	rxhtmlTag = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([\w:]+)[^>]*)\/>/gi,
-	rtagName = /<([\w:]+)/,
+	rtagName = /<([\w:-]+)/,
```

Comment:
`rsingleTag` is the test for element creation via [`document.createElement`](https://github.com/jquery/jquery/blob/740e190223d19a114d5373758127285d14d6b71e/src/core/parseHTML.js#L21-L27), and therefore shouldn't include ":".

`rtagName` is for [finding the first open tag in an HTML string](https://github.com/jquery/jquery/blob/740e190223d19a114d5373758127285d14d6b71e/src/manipulation.js#L220), and therefore _should_ include ":" (as should its partner `rxhtmlTag`).

All of the above should probably find its way into source comments.